### PR TITLE
Clean-up some (not all) fencing hacks

### PR DESCRIPTION
**Pull request by abays**
Removes some dev-tools Kubevirt fencing hacks.

### CODE REVIEW

File: ansible/templates/osp/tripleo_fencing_overrides.yaml.j2 [hunk 1]
```diff
@@ -8,17 +8,6 @@
       #!/bin/bash
       /usr/libexec/platform-python -m pip install openshift
 
-  # Note: can be removed when is available https://review.opendev.org/q/topic:%22fence_kubevirt%22
-  - name: Pre-install certain packaged
-    package:
-      state: installed
-      name: "{{ '{{' }} item {{ '}}' }}"
-    with_items:
-      - puppet-tripleo
```
Comment (stuggi) [inline ansible/templates/osp/tripleo_fencing_overrides.yaml.j2:17]: for 16.2 we'd still need the puppet-tripleo patch I recently backported in https://review.opendev.org/c/openstack/puppet-tripleo/+/810682

Comment (stuggi) [reply ansible/templates/osp/tripleo_fencing_overrides.yaml.j2:17]: should we keep this and make a condition depending on the OSP version, or just wait for a compose which has it available?

File: ansible/templates/osp/tripleo_fencing_overrides.yaml.j2 [hunk 2]
```diff
@@ -39,25 +28,3 @@
...
-      if ! curl https://review.opendev.org/changes/openstack%2Fpuppet-pacemaker~806924/revisions/1/patch?download | base64 -d | filterdiff -p1 -x 'spec/*' | patch -R -s -f --dry-run -d /usr/share/openstack-puppet/modules/pacemaker -p1; then
-        curl https://review.opendev.org/changes/openstack%2Fpuppet-pacemaker~806924/revisions/1/patch?download | base64 -d | filterdiff -p1 -x 'spec/*' | patch -d /usr/share/openstack-puppet/modules/pacemaker -b -z .orig -p1
-      fi
-      if ! curl https://review.opendev.org/changes/openstack%2Fpuppet-pacemaker~807299/revisions/1/patch?download | base64 -d | filterdiff -p1 -x 'spec/*' | patch -R -s -f --dry-run -d /usr/share/openstack-puppet/modules/pacemaker -p1; then
-        curl https://review.opendev.org/changes/openstack%2Fpuppet-pacemaker~807299/revisions/1/patch?download | base64 -d | filterdiff -p1 -x 'spec/*' | patch -d /usr/share/openstack-puppet/modules/pacemaker -b -z .orig -p1
-      fi
-      if ! curl https://review.opendev.org/changes/openstack%2Fpuppet-tripleo~806926/revisions/2/patch?download | base64 -d | filterdiff -p1 -x 'spec/*' | patch -R -s -f --dry-run -d /usr/share/openstack-puppet/modules/tripleo -p1; then
```
Comment (stuggi) [inline ansible/templates/osp/tripleo_fencing_overrides.yaml.j2:60]: same here we might want to keep this for OSP16 and use https://review.opendev.org/c/openstack/puppet-tripleo/+/810682 ?

File: ansible/templates/osp/tripleo_fencing_overrides.yaml.j2 (hunk 1 above)
Comment (abays) [reply ansible/templates/osp/tripleo_fencing_overrides.yaml.j2:17]: Ah, okay.  When I last tested this, I must have tried with 17 and saw it was there.  Let's just leave this in for now, then.  Your patch is available, we can finally remove these pieces in `tripleo_fencing_overrides.yaml.j2`.

File: ansible/templates/osp/tripleo_fencing_overrides.yaml.j2 (hunk 2 above)
Comment (stuggi) [reply ansible/templates/osp/tripleo_fencing_overrides.yaml.j2:60]: I think we can remove some of them puppet-pacemaker should be there. for puppet-tripleo I don't think 806926 won't apply clean on 16.2. we'd need to use 810682

File: ansible/templates/osp/tripleo_fencing_overrides.yaml.j2 (hunk 1 above)
Comment (abays) [reply ansible/templates/osp/tripleo_fencing_overrides.yaml.j2]: Can we leave this block as-is and just use conditional stuff for the patches in the other block below?

Comment (stuggi) [reply ansible/templates/osp/tripleo_fencing_overrides.yaml.j2]: yes, it won't hurt to install the packages. they get installed anyway.

File: ansible/templates/osp/tripleo_fencing_overrides.yaml.j2 (hunk 2 above)
Comment (abays) [reply ansible/templates/osp/tripleo_fencing_overrides.yaml.j2:60]: I _think_ I made the changes you suggested.  When you have time, can you take a look?

File: ansible/templates/osp/tripleo_fencing_overrides.yaml.j2 [hunk 3]
```diff
@@ -57,7 +51,12 @@
...
+      if [[ "{{ osp.release }}" == "16.2" ]]; then
+        if ! curl https://review.opendev.org/changes/openstack%2Fpuppet-tripleo~810682/revisions/1/patch?download | base64 -d | filterdiff -p1 -x 'spec/*' | patch -R -s -f --dry-run -d /usr/share/openstack-puppet/modules/tripleo -p1; then
+          curl https://review.opendev.org/changes/openstack%2Fpuppet-tripleo~810682/revisions/1/patch?download | base64 -d | filterdiff -p1 -x 'spec/*' | patch -d /usr/share/openstack-puppet/modules/tripleo -b -z .orig -p1
+        fi
+      else
+        if ! curl https://review.opendev.org/changes/openstack%2Fpuppet-tripleo~806926/revisions/2/patch?download | base64 -d | filterdiff -p1 -x 'spec/*' | patch -R -s -f --dry-run -d /usr/share/openstack-puppet/modules/tripleo -p1; then
+          curl https://review.opendev.org/changes/openstack%2Fpuppet-tripleo~806926/revisions/2/patch?download | base64 -d | filterdiff -p1 -x 'spec/*' | patch -d /usr/share/openstack-puppet/modules/tripleo -b -z .orig -p1
+        fi
```
Comment (stuggi) [inline ansible/templates/osp/tripleo_fencing_overrides.yaml.j2:61]: we do not required this as this is already in the puppet-tripleo rpm for 17.0 . so we'd only need the 16.2 one above

Comment (stuggi) [reply ansible/templates/osp/tripleo_fencing_overrides.yaml.j2:61]: from rpm changelog

```
* Sat Sep 04 2021 OSP Prod Chain <dev-null@redhat.com> 15.0.1-0.20210905003900.2addd1d
- Update patches
- Extend fencing to hosts using fence_kubevirt agent.
```

**Review by stuggi — APPROVED** [review]
/lgtm

**Pull request merged**